File: 프로그래머스/1/42840. 모의고사/모의고사.swift
import Foundation

func solution(_ answers:[Int]) -> [Int] {
    let s1 = [1, 2, 3, 4, 5]
    let s2 = [2, 1, 2, 3, 2, 4, 2, 5]
    let s3 = [3, 3, 1, 1, 2, 2, 4, 4, 5, 5]
    var result: [Int] = []
    var scores: [Int] = [0, 0, 0]
    
    for (num, answer) in answers.enumerated() {
        if s1[num % s1.count] == answer { scores[0] += 1 }
        if s2[num % s2.count] == answer { scores[1] += 1 }
        if s3[num % s3.count] == answer { scores[2] += 1 }
    }
    
    var max = scores.sorted(by: >)[0]
    
    for (index, score) in scores.enumerated() {
        if max == score {
            result.append(index+1)
        }
    }
    
    return result
}

File: 프로그래머스/1/42840. 모의고사/모의고사.cpp
#include <algorithm>
#include <iostream>
#include <vector>

using namespace std;

vector<int> solution(vector<int> answers) {
    vector<int> answer, result(3, 0);
    int arr1[5] = {1, 2, 3, 4, 5};
    int arr2[8] = {2, 1, 2, 3, 2, 4, 2, 5};
    int arr3[10] = {3, 3, 1, 1, 2, 2, 4, 4, 5, 5};
    int maxScore = 0;

    for (int i = 0; i < answers.size(); i++) {
        if (answers[i] == arr1[i % 5]) {
            result[0]++;
        }
        if (answers[i] == arr2[i % 8]) {
            result[1]++;
        }
        if (answers[i] == arr3[i % 10]) {
            result[2]++;
        }
    }

    maxScore = *max_element(result.begin(), result.end());

    for (int i = 0; i < 3; i++) {
        if (maxScore==result[i]) {
            answer.push_back(i + 1);
        }
    }

    return answer;
}